src/journal/JournalRecorder.cc

```diff
@@ -262,7 +279,7 @@ void JournalRecorder::create_next_object_recorder(
       new_object_recorder->get_object_number());
   }
 
-  new_object_recorder->append(append_buffers);
+  new_object_recorder->append(append_buffers, false);
```

Comment:
Minor: just to keep the locking consistent, how about storing the new recorder, acquiring the splay offset lock, releasing ```m_lock```, and then invoking ```append_unlock```